app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Edit.php

```diff
@@ -6,53 +6,72 @@
+use Magento\Framework\App\ObjectManager;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\View\Result\PageFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Eav\Api\AttributeSetRepositoryInterface;
+use Magento\Catalog\Controller\Adminhtml\Product\Set;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 
-class Edit extends \Magento\Catalog\Controller\Adminhtml\Product\Set implements HttpGetActionInterface
+class Edit extends Set implements HttpGetActionInterface
 {
     /**
-     * @var \Magento\Framework\View\Result\PageFactory
+     * @var PageFactory
      */
     protected $resultPageFactory;
 
     /**
-     * @param \Magento\Backend\App\Action\Context $context
-     * @param \Magento\Framework\Registry $coreRegistry
-     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
+     * @var AttributeSetRepositoryInterface
+     */
+    protected $attributeSetRepository;
+
+    /**
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param PageFactory $resultPageFactory
+     * @param AttributeSetRepositoryInterface $attributeSetRepository
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Framework\Registry $coreRegistry,
-        \Magento\Framework\View\Result\PageFactory $resultPageFactory
+        Context $context,
+        Registry $coreRegistry,
+        PageFactory $resultPageFactory,
+        AttributeSetRepositoryInterface $attributeSetRepository = null
     ) {
         parent::__construct($context, $coreRegistry);
         $this->resultPageFactory = $resultPageFactory;
+        $this->attributeSetRepository = $attributeSetRepository ?:
+            ObjectManager::getInstance()->get(AttributeSetRepositoryInterface::class);
     }
 
     /**
-     * @return \Magento\Backend\Model\View\Result\Page
+     * @return ResultInterface
+     *
+     * @throws NoSuchEntityException
      */
     public function execute()
     {
         $this->_setTypeId();
-        $attributeSet = $this->_objectManager->create(\Magento\Eav\Model\Entity\Attribute\Set::class)
-            ->load($this->getRequest()->getParam('id'));
-
+        $attributeSet = $this->attributeSetRepository->get($this->getRequest()->getParam('id'));
         if (!$attributeSet->getId()) {
-            return $this->resultRedirectFactory->create()->setPath('catalog/*/index');
+            return $this->resultFactory->create(ResultFactory::TYPE_REDIRECT)->setPath('catalog/*/index');
         }
-
         $this->_coreRegistry->register('current_attribute_set', $attributeSet);
 
-        /** @var \Magento\Backend\Model\View\Result\Page $resultPage */
-        $resultPage = $this->resultPageFactory->create();
+        /** @var Page $resultPage */
+        $resultPage = $this->resultFactory->create(ResultFactory::TYPE_PAGE);
```

Comment:
Same here - there is no recommendation to not use typified factories like `$this->resultPageFactory`. As to me they are more convenient syntactic sugar and make IDE type hinting work properly as well.

Reply:
The PHPDoc annotation would would help with the IDE type hinting, right?

Reply:
Sure. But it's additional effort to add it. Note that in current implementation such PHPDoc is not even needed.

Approving current changeset and in regards to getting rid of `$this->resultRedirectFactory` let's wait for Ihor PR discussed with architects. Then we could do massive cleanup.

Reply:
@orlangur Alright, sounds great. Thanks!